Assets/Scripts/PlayerController.cs

```diff
@@ -124,23 +134,55 @@ private Vector3 GetVector3FromString(string teleport)
                                        .FirstOrDefault();
     }
 
-    private void FadeUI(string teleportTo)
+    private void FadeUITeleport(
+        string teleportTo, 
+        bool newResetPostition = false,
+        bool enteringMainHub = false)
     {
-        StartCoroutine(DoFadeUI(teleportTo));
+        StartCoroutine(DoFadeUITeleport(teleportTo, newResetPostition, enteringMainHub));
     }
 
-    private IEnumerator DoFadeUI(string teleportTo)
+    private void FadeUIReset()
     {
+        StartCoroutine(DoFadeUIReset());
+    }
+
+    private IEnumerator DoFadeUITeleport(string teleportTo, bool newResetPostition, bool enteringMainHub)
+    {
+        disableControls = true;
         teleportUIAnimator.SetTrigger("StartTeleport");
 
-        yield return new WaitForSeconds(1);
+        yield return new WaitForSeconds(0.5f);
 
         var vector3NewPosition = GetVector3FromString(teleportTo);
         playerRb.position = vector3NewPosition;
-        currentPlayerResetPosition = vector3NewPosition;
         ResetPlayer();
-        inMainHub = !inMainHub;
+        inMainHub = enteringMainHub;
```

Comment:
No need to send enteringMainHub boolean from earlier methods. Could just be:
`inMainHub = teleportTo.equals("MainHub")`

Reply:
Fixed